agent_api/src/main/java/dev/aikido/agent_api/storage/ServiceConfiguration.java

```diff
@@ -64,21 +71,80 @@ public void setMiddlewareInstalled(boolean middlewareInstalled) {
         return endpoints;
     }
 
-    public HashSet<String> getBlockedUserIDs() {
-        return blockedUserIDs;
+    public boolean isUserBlocked(String userId) {
+        return this.blockedUserIDs.contains(userId);
+    }
+    public boolean isIpBypassed(String ip) {
+        return this.bypassedIPs.matches(ip);
+    }
+
+    /**
+     * Check if the IP is blocked (e.g. Geo IP Restrictions)
+     */
+    public BlockedResult isIpBlocked(String ip) {
+        // Check for allowed ip addresses (i.e. only one country is allowed to visit the site)
+        // Always allow access from private IP addresses (those include local IP addresses)
+        if (!allowedIps.isEmpty() && !isPrivateIp(ip)) {
+            boolean ipAllowed = false;
+            for (IPListEntry entry : allowedIps) {
+                if (entry.ipList.matches(ip)) {
+                    ipAllowed = true; // We allow IP addresses as long as they match with one of the lists.
+                    break;
+                }
+            }
+            if (!ipAllowed) {
+                return new BlockedResult(true, "not in allowlist");
+            }
+        }
+
+        // Check for blocked ip addresses
+        for (IPListEntry entry : blockedIps) {
+            if (entry.ipList.matches(ip)) {
+                return new BlockedResult(true, entry.description);
+            }
+        }
+        return new BlockedResult(false, null);
+    }
+
+    public void updateBlockedLists(ReportingApi.APIListsResponse res) {
+        // Update blocked IP addresses (e.g. for geo restrictions) :
+        if (res.blockedIPAddresses() != null) {
+            for (ReportingApi.ListsResponseEntry entry : res.blockedIPAddresses()) {
+                IPList ipList = createIPList(entry.ips());
+                blockedIps.add(new IPListEntry(ipList, entry.description()));
+            }
+        }
+        // Update allowed IP addresses (e.g. for geo restrictions) :
+        if (res.allowedIPAddresses() != null) {
+            for (ReportingApi.ListsResponseEntry entry : res.allowedIPAddresses()) {
+                IPList ipList = createIPList(entry.ips());
+                this.allowedIps.add(new IPListEntry(ipList, entry.description()));
+            }
+        }
+        // Update Blocked User-Agents regex
+        if (res.blockedUserAgents() != null && !res.blockedUserAgents().isEmpty()) {
+            this.blockedUserAgentRegex = Pattern.compile(res.blockedUserAgents(), Pattern.CASE_INSENSITIVE);
+        }
     }
 
-    public void storeBlockedListsRes(Optional<ReportingApi.APIListsResponse> apiListsResponse) {
-        if (apiListsResponse.isPresent()) {
-            this.blockedListsRes = apiListsResponse.get();
+    /**
+     * Check if a given User-Agent is blocked or not :
+     */
+    public boolean isBlockedUserAgent(String userAgent) {
+        if (blockedUserAgentRegex != null) {
+            return blockedUserAgentRegex.matcher(userAgent).find();
```

Comment:
```suggestion
            return blockedUserAgentRegex.matcher(userAgent).matches();
```

`find` works the same but express less the intent?

Reply:
matches tries to match the entire string, which we don't want, otherwise the bot blocking would be broken